lib/Sema/TypeCheckPattern.cpp

```diff
@@ -736,6 +736,74 @@ static bool validateParameterType(ParamDecl *decl, DeclContext *DC,
     }
     decl->getTypeLoc().setType(Ty);
   }
+  // If the param is not a 'let' and it is not an 'inout'.
+  // It must be a 'var'. Provide helpful diagnostics like a shadow copy
+  // in the function body to fix the 'var' attribute.
+  if (!decl->isLet() && !Ty->is<InOutType>()) {
+    auto innermostMethodCtx = dyn_cast_or_null<AbstractFunctionDecl>(DC);
+    if (!innermostMethodCtx || !innermostMethodCtx->hasBody()) {
+      // If there is not function body, just suggest removal of 'var'
+      TC.diagnose(decl->getLetVarInOutLoc(), diag::var_parameter_not_allowed)
+      .fixItRemove(decl->getLetVarInOutLoc());
+    } else {
+      // Insert the shadow copy. The computations that follow attempt to
+      // 'best guess' the indentation and new lines so that the user
+      // doesn't have to add any whitespace.
+      auto declBody = innermostMethodCtx->getBody();
+      
+      auto &SM = TC.Context.SourceMgr;
+      
+      SourceLoc insertionStartLoc;
+      std::string start;
+      std::string end;
+      
+      auto lBraceLine = SM.getLineNumber(declBody->getLBraceLoc());
+      
+      if (!declBody->getNumElements()) {
+        // emtpy function body
+        auto rBraceLine = SM.getLineNumber(declBody->getRBraceLoc());
```

Comment:
I'd suggest hoisting this call out of the if since both arms need it.  Doing so would increase readability of the code by making the left/right side more analogous.